tests/docker_mcp/test_mcp.py

```
@@ -1,1304 +1,588 @@
+        except subprocess.TimeoutExpired:
+            process.kill()
+
+
+def read_json_line(process, timeout_seconds=10):
+    deadline = time.time() + timeout_seconds
+    while time.time() < deadline:
         line = process.stdout.readline()
         if not line:
-            return {"error": "no response"}
+            stderr = process.stderr.read()
+            raise RuntimeError(f"mcp server closed stdout early; stderr={stderr!r}")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)